src/CollisionDetection.hs

```diff
@@ -1,55 +1,77 @@
 module CollisionDetection where
```

Comment:
You probably want an explicit export list, to avoid accidentally exposing private definitions.